SchoolProject/src/main/java/Menus/TeacherMenu.java

```diff
@@ -0,0 +1,155 @@
+package Menus;
+
+import java.util.List;
+import java.util.Scanner;
+
+import Helpers.EducationLevel;
+import Helpers.Grade;
+import Models.Student;
+import Models.Subject;
+import Models.Teacher;
+import Repositories.StudentRepository;
+import Repositories.TeacherRepository;
+import Services.AccessValidator;
+
+public class TeacherMenu implements IMenuCommand{
+
+    private TeacherRepository teachers;
+    private StudentRepository students;
+    private String studentCode;
+
+    public TeacherMenu(String studentCode) {
+        this.studentCode = studentCode;
+    }
+
+    @Override
+    public void execute() {
+        Scanner scanner = new Scanner(System.in);
+        teachers = new TeacherRepository();
+        Teacher teacher = teachers.getTeacherByCode(studentCode);
+        System.out.println("=============================");
+        System.out.println("Welcome " + teacher.getName());
+        System.out.println("=============================");
+        System.out.println("Select your option please: ");
+        System.out.println("1. List all students by Education Level");
+        System.out.println("2. Enter a grade for a student");
+        System.out.println("3. Exit");
+        String option = scanner.nextLine();
+        boolean access = false;
+        while (!access) {
+            switch (option) {
+                case "1":
+                    menuOption1();
```

Comment:
you can divide the responsibilities here, maybe a static class that only display static messages, just to display the principal menu